code/modules/projectiles/gun_attachables.dm

```diff
@@ -249,6 +249,7 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	for(var/datum/action/action_to_update AS in master_gun.actions)
 		if(action_to_update.target != src)
 			continue
+		LAZYREMOVE(master_gun.actions, action_to_update)
 		qdel(action_to_update)
```

Comment:
why is this here of all places

Reply:
you mean the qdel ? No idea

Reply:
No why did you add the lazyremove

Reply:
Why would it not remove it? even if it doesnt why wouldnt it be handled in the Destroy()